pandas/core/strings.py

```diff
@@ -1815,36 +1817,34 @@ def __init__(self, data):
     def _validate(data):
         from pandas.core.index import Index
 
-        if (isinstance(data, ABCSeries) and
-                not ((is_categorical_dtype(data.dtype) and
-                      is_object_dtype(data.values.categories)) or
-                     (is_object_dtype(data.dtype)))):
-            # it's neither a string series not a categorical series with
-            # strings inside the categories.
-            # this really should exclude all series with any non-string values
-            # (instead of test for object dtype), but that isn't practical for
-            # performance reasons until we have a str dtype (GH 9343)
+        if isinstance(data, ABCMultiIndex):
+                raise AttributeError('Can only use .str accessor with Index, '
+                                     'not MultiIndex')
+
+        # see src/inference.pyx which can contain string values
+        allowed_types = ['string', 'unicode', 'empty',
+                         'mixed', 'mixed-integer']
+        if isinstance(data, ABCSeries):
```

Comment:
why does this accept bytes?

Reply:
See OP:
> What needs disussion is the following: I've commented out two tests about an encode-decode-roundtrip that cannot work when forbidding `bytes` data. Should such an encode-decode-cycle explicitly part of the desired functionality for `.str` (currently there are `.str.encode`  and `.str.decode` methods), or not? I guess that even if not, this would need a deprecation cycle.

I originally commented out the only tests in `test_strings` that needed `bytes` and completely forbade `bytes`, but then there were lots of test errors, see e.g. https://travis-ci.org/pandas-dev/pandas/jobs/441825774
Alternatively, it'd be possible to allow construction of the `StringMethods` object but raise on `inferred_dtype='bytes'` for all but the `str.decode` method.